### PR TITLE
feat!: FlatRuleTester -> RuleTester

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[x] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

1. Removed `RuleTester`
1. Renamed `FlatRuleTester` to `RuleTester`
1. Updated `api.js` to export the new `RuleTester`
2. Updated `unsupported-api.js` to stop exporting `FlatRuleTester`
3. Updated the API docs
4. Updated the Migrate to v9 guide
5. Updated all rule tests

### CODE REVIEW

**Review by bmish — APPROVED** [review]

File: docs/src/use/migrate-to-9.0.0.md [hunk 1]
```diff
@@ -158,6 +159,56 @@ The [eslint-plugin/require-meta-schema](https://github.com/eslint-community/esli
...
+            ecmaVersion: 5,
+            sourceType: "script"
+        }
+    });
+    ```
+
+* **Change `parserOptions` to `languageOptions`.** If you're setting `ecmaVersion` or `sourceType` on your tests, move those from `parserOptions` to `languageOptions`, like this:
+
+    ```js
+    ruleTester.run({
```
Comment (mdjermanovic) [inline docs/src/use/migrate-to-9.0.0.md:183]:
```suggestion
    ruleTester.run("my-rule", myRule, {
```

File: docs/src/use/migrate-to-9.0.0.md [hunk 2]
```diff
@@ -158,6 +159,56 @@ The [eslint-plugin/require-meta-schema](https://github.com/eslint-community/esli
...
+                parserOptions: {
+                    ecmaVersion: 6
+                }
+            }
+        ]
+    });
+
+    // becomes
+
+    ruleTester.run({
```
Comment (mdjermanovic) [inline docs/src/use/migrate-to-9.0.0.md:196]:
```suggestion
    ruleTester.run("my-rule", myRule, {
```

File: docs/src/use/migrate-to-9.0.0.md [hunk 3]
```diff
@@ -158,6 +159,56 @@ The [eslint-plugin/require-meta-schema](https://github.com/eslint-community/esli
...
+                code: "foo",
+                languageOptions: {
+                    ecmaVersion: 6
+                }
+            }
+        ]
+    });
+    ```
+
+* **Translate other config keys.** Keys such as `env` and `parser` that used to run on the eslintrc config system must be translated into the flat config system. Please refer to the [Configuration Migration Guide](/docs/latest/use/configure/migration-guide) for details on translating other keys you may be using.
```
Comment (mdjermanovic) [inline docs/src/use/migrate-to-9.0.0.md:208]:
```suggestion
* **Translate other config keys.** Keys such as `env` and `parser` that used to run on the eslintrc config system must be translated into the flat config system. Please refer to the [Configuration Migration Guide](configure/migration-guide) for details on translating other keys you may be using.
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**